### PR TITLE
BTS-1292 Cleanup dangling links (#18521)

### PR DESCRIPTION
**Pull request by Dronplane**
Backport of https://github.com/arangodb/arangodb/pull/18521 and https://github.com/arangodb/arangodb/pull/18599

### CODE REVIEW

File: arangod/Cluster/ClusterTypes.h [hunk 1]
```diff
@@ -30,6 +30,9 @@
 #include <string>
 
 #include "Basics/Result.h"
+==== BASE ====
+#include "Containers/FlatHashMap.h"
+==== BASE ====
```
Comment (jsteemann) [inline arangod/Cluster/ClusterTypes.h:35]: This looks weird.

File: arangod/Cluster/RebootTracker.cpp [hunk 2]
```diff
@@ -203,4 +202,10 @@ void RebootTracker::unregisterCallback(RebootTracker::PeerState const& peer,
   }
 }
 
+bool RebootTracker::isServerAlive(ServerID id) const {
```
Comment (jsteemann) [inline arangod/Cluster/RebootTracker.cpp:205]:
```suggestion
bool RebootTracker::isServerAlive(ServerID const& id) const {
```

File: arangod/Cluster/RebootTracker.h [hunk 3]
```diff
@@ -73,7 +72,9 @@ class RebootTracker {
   CallbackGuard callMeOnChange(PeerState peer, Callback callback,
                                std::string description);
 
-  void updateServerState(State state);
+  void updateServerState(ServersKnown state);
+
+  bool isServerAlive(ServerID id) const;
```
Comment (jsteemann) [inline arangod/Cluster/RebootTracker.h:77]:
```suggestion
  bool isServerAlive(ServerID const& id) const;
```

File: arangod/Cluster/ClusterInfo.cpp [hunk 4]
```diff
@@ -261,6 +264,131 @@ inline arangodb::AgencyOperation CreateCollectionSuccess(
...
+    VPackSlice const serversKnownSlice, VPackSlice const supervisionHealth,
+    containers::FlatHashSet<ServerID> const& serverIds) {
+  ServersKnown res;
+  TRI_ASSERT(serversKnownSlice.isNone() || serversKnownSlice.isObject());
+  LOG_TOPIC("91da8", TRACE, Logger::CLUSTER)
+      << "Supervision health is:" << supervisionHealth.toString();
+  if (serversKnownSlice.isObject()) {
+    for (auto const it : VPackObjectIterator(serversKnownSlice)) {
+      auto status = ServerHealth::kUnclear;
+      std::string serverId = it.key.copyString();
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp:312]:
```suggestion
      std::string_view serverId = it.key.stringView();
```

File: arangod/Cluster/ClusterTypes.h (hunk 1 above)
Comment (Dronplane) [reply arangod/Cluster/ClusterTypes.h:35]: merge artifact. fixed

File: arangod/Cluster/ClusterInfo.cpp (hunk 4 above)
Comment (Dronplane) [reply arangod/Cluster/ClusterInfo.cpp:312]: We anyway will need a string copy here. As at line 336 we will move this string into result. So we can have a string_view here, but still later we will copy it into result.  Wit that assumption I think copyString here is fine.

Comment (jsteemann) [reply arangod/Cluster/ClusterInfo.cpp:285]: Ok

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by MBkkt — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]

**Pull request merged**